R/ui.R

```diff
@@ -5,47 +5,66 @@
 #' \url{https://github.com/r-lib/usethis/blob/master/R/ui.R}.
 #' These console messages are created such that the user is
 #' aware of the effects of removing specific components from
-#' the modeling object.
+#' the model object.
 #'
-#' @param og original model object
-#' @param butchered butchered model object
+#' @param og Original model object.
+#' @param butchered Butchered model object.
 #'
 #' @name ui
 
 #' @rdname ui
-#' @return pretty version of memory released
 memory_released <- function(og, butchered) {
   old <- lobstr::obj_size(og)
   new <- lobstr::obj_size(butchered)
   rel <- old - new
   rel <- format(rel, big.mark = ",", scientific = FALSE)
-  if(length(rel) == 1) {
-    return(paste(rel, "B"))
+  if (length(rel) == 1) {
+    if (rel <= 0) {
+      return(NULL)
+    } else {
+      return(paste(rel, "B"))
+    }
   } else {
-    return(paste("0", "B"))
+    return(NULL)
   }
 }
 
 #' @rdname ui
-#' @param working string of tested working functions
-#' @param broken string of nonworking functions
-#' @param class_added boolean default to true when butcher class appended
-#' @return console messages
+#' @param disabled String of disabled functions as a result of axing.
+#' @param class_added Boolean that defaults to true when butcher class is added.
 assess_object <- function(og, butchered,
-                          working = NULL,
-                          broken = NULL,
+                          disabled = NULL,
                           class_added = TRUE) {
   mem <- memory_released(og, butchered)
-  ui_done("Memory released: {ui_value(mem)}")
-  if(!is.null(working)) {
-    ui_done("Active: {ui_code(working)}")
-  }
-  if(!is.null(broken)) {
-    ui_oops("Inactive: {ui_code(broken)}")
-  }
-  if(!class_added) {
-    class_name <- "butchered"
-    ui_oops("Could not append additional {ui_value(class_name)} class")
+  if (is.null(mem)) {
+    ui_oops("No memory released. Do not butcher.")
+  } else {
+    ui_done("Memory released: {ui_value(mem)}")
+    if (!is.null(disabled)) {
+      ui_oops("Disabled: {ui_code(disabled)}")
```

Comment:
Would it make sense to have some kind of verbage saying that _at least_ these things have been disabled, but there might be other things that don't work anymore that we aren't sure about?

Reply:
good point! ive been testing every function specific to each modeling package on the axed model object, but haven't tested functions outside of the package that might use the axed modeling object. will think of a concise way to change the error message and detail this issue in the documentation